src/creature.h

```diff
@@ -235,6 +235,7 @@ class Creature
         // increase pain, apply effects, etc
         virtual void apply_damage(Creature *source, body_part bp, int amount) = 0;
 
+        virtual void on_damage_taken( Creature *source, dealt_damage_instance &d ) = 0; // if this receives actual damage, do something with it
```

Comment:
This is exactly the same meaning as `on_hurt`.
Either merge those two or get rid of this from `Creature` class and move it only to `monster`.